lib/rucio/core/rse.py

```diff
@@ -520,6 +557,43 @@ def get_rse_attribute(key, rse_id=None, value=None, session=None):
     return result
 
 
+@read_session
+def get_rse_supported_checksums(rse_id=None, session=None):
+    """
+    Retrieve RSE attribute value.
+
+    :param rse_id: The RSE id.
+    :param session: The database session in use.
+
+    :returns: The list of checksums supported by the selected RSE.
+    """
+
+    # Collect enabled and disabled checksums
+    checksum_attr_list = list(filter(lambda x: (get_rse_attribute(get_checksum_attr_key(x), rse_id=rse_id, session=session) is not None), GLOBALLY_SUPPORTED_CHECKSUMS))
```

Comment:
can't see it from the diff, but if this is a core method you need to pass through the session to get_rse_attribute  (also in the call below)

Reply:
Nice point. I see the way that decorator works, but how should I change the arguments of the `get_rse_attribute`? Isn't session=session performing a passthrough? Sorry for the naive question :)

Reply:
It seems to me I should still add session=None in the args, but then a session global-like variable will be generated and passed to the inner method, right?